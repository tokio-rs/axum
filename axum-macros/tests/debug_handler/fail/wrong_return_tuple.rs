#![allow(unused_parens)]

#[axum::debug_handler]
async fn named_type() -> (
    axum::http::StatusCode,
    axum::Json<&'static str>,
    axum::response::AppendHeaders<[(axum::http::HeaderName, &'static str); 1]>,
) {
    panic!()
}

struct CustomIntoResponse {}
impl axum::response::IntoResponse for CustomIntoResponse {
    fn into_response(self) -> axum::response::Response {
        todo!()
    }
}
#[axum::debug_handler]
async fn custom_type() -> (
    axum::http::StatusCode,
    CustomIntoResponse,
    axum::response::AppendHeaders<[(axum::http::HeaderName, &'static str); 1]>,
) {
    panic!()
}

fn main() {}
