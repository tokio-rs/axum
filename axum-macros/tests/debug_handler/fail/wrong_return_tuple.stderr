error: `Json<_>` must be the last element in a response tuple
 --> tests/debug_handler/fail/wrong_return_tuple.rs:6:5
  |
6 |     axum::Json<&'static str>,
  |     ^^^^^^^^^^^^^^^^^^^^^^^^

error[E0277]: the trait bound `CustomIntoResponse: IntoResponseParts` is not satisfied
  --> tests/debug_handler/fail/wrong_return_tuple.rs:24:5
   |
24 |     CustomIntoResponse,
   |     ^^^^^^^^^^^^^^^^^^ the trait `IntoResponseParts` is not implemented for `CustomIntoResponse`
   |
   = help: the following other types implement trait `IntoResponseParts`:
             AppendHeaders<I>
             HeaderMap
             Extension<T>
             Extensions
             Option<T>
             [(K, V); N]
             ()
             (T1,)
           and $N others
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable

error[E0277]: the trait bound `CustomIntoResponse: IntoResponseParts` is not satisfied
  --> tests/debug_handler/fail/wrong_return_tuple.rs:24:5
   |
24 |     CustomIntoResponse,
   |     ^^^^^^^^^^^^^^^^^^ the trait `IntoResponseParts` is not implemented for `CustomIntoResponse`
   |
   = help: the following other types implement trait `IntoResponseParts`:
             AppendHeaders<I>
             HeaderMap
             Extension<T>
             Extensions
             Option<T>
             [(K, V); N]
             ()
             (T1,)
           and $N others
note: required by a bound in `__axum_macros_check_custom_type_into_response_parts_1_check`
  --> tests/debug_handler/fail/wrong_return_tuple.rs:24:5
   |
24 |     CustomIntoResponse,
   |     ^^^^^^^^^^^^^^^^^^ required by this bound in `__axum_macros_check_custom_type_into_response_parts_1_check`
